Godeps/_workspace/src/k8s.io/kubernetes/pkg/kubectl/rolling_updater.go

```diff
@@ -307,14 +307,14 @@ func (r *RollingUpdater) scaleDown(newRc, oldRc *api.ReplicationController, desi
 		return oldRc, nil
 	}
 	// Block until there are any pods ready.
-	_, newAvailable, err := r.waitForReadyPods(config.Interval, config.Timeout, oldRc, newRc)
+	oldAvailable, newAvailable, err := r.waitForReadyPods(config.Interval, config.Timeout, oldRc, newRc)
 	if err != nil {
 		return nil, err
 	}
 	// The old controller is considered as part of the total because we want to
 	// maintain minimum availability even with a volatile old controller.
 	// Scale down as much as possible while maintaining minimum availability
-	decrement := oldRc.Spec.Replicas + newAvailable - minAvailable
+	decrement := oldAvailable + newAvailable - minAvailable
```

Comment:
We also need to take into account unavailable old pods and scale those down (https://github.com/openshift/origin/issues/7271) ........